### PR TITLE
Fix/remove default props/android layout drawer

### PR DESCRIPTION
**Pull request by mdaj06**
## Summary
Removed the deaultProps in the DrawerLayoutAndroid file and replaced it with a default value in case props are undefined.

## Changelog
[General] [Changed] - Remove defaultProps from the DrawerLayoutAndroid Component.
@lunaleaps this is the fix for issue #31606

## Test Plan
Ran test suite

### CODE REVIEW

File: Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js [hunk 1]
```diff
@@ -45,19 +45,6 @@ type Props = $ReadOnly<{|
...
-   * If you want to set the opacity of the drawer, use rgba. Example:
-   *
-   * ```
-   * return (
-   *   <DrawerLayoutAndroid drawerBackgroundColor="rgba(0,0,0,0.5)">
-   *   </DrawerLayoutAndroid>
-   * );
-   * ```
-   */
-  drawerBackgroundColor: ColorValue,
```
Comment (mdaj06) [inline Libraries/Components/DrawerAndroid/DrawerLayoutAndroid.android.js:59]: @lunaleaps i think we can remove this because, when a user passes a custom prop to the component it will pick it up and replace the default value which is currently - "white". if we keep it here it would raise an error saying that drawerBackgroundColor is present in "Props"(read only flow interface) and not present inside the props in the DrawerLayoutAndroid Class component. 

When the "static props" field was present this wasn't an issue as the class by default added it to the props object.

File: gradlew.bat [hunk 2]
```diff
@@ -1,89 +1,89 @@
-@rem
```
Comment (cortinico) [inline gradlew.bat:1]: Similarly to #32133, you should not add `gradlew.bat` files to this PR (please don't `git add --all` otherwise you will add those files).

Comment (mdaj06) [inline gradlew.bat:1]: Regarding the .bat files, they were present from the moment i did a rebase from the "main" branch essentially. if you compare the left and right panes there are no changes. So this shouldn't affect the current code, but it is weird that i could not stash/revert these changes after a merge from the  upstream main branch.

Comment (cortinico) [reply gradlew.bat:1]: There are indeed changes in terms of whitespaces. This is addressed by #31398 and you should not include them in your change as they're unrelated to your PR.

Comment (mdaj06) [reply gradlew.bat:1]: yes doing a git add <filename> would fix this, but wasnt sure as it appeared after i took a pull form the main branch. so shall i reverse the commit and only push the other relevant files? @cortinico . Not too sure with the exact reason why this kept showing up after the pull.

Comment (mdaj06) [reply gradlew.bat:1]: got it! makes sense thanks for pointing this out 👍🏼

**Comment by cortinico** [comment]
The diff for this PR seems to be empty @mdaj06

**Comment by mdaj06** [comment]
@cortinico this branch got closed on a revert so i have addressed all the issues mentioned above in PR #32162

**Comment by cortinico** [comment]
Gotcha, sorry for the confusion

File: gradlew.bat (hunk 2 above)
Comment (fabriziobertoglio1987) [reply gradlew.bat]: https://github.com/fabriziobertoglio1987/react-native-notes/issues/8#issuecomment-1043856916
